torch/_inductor/codegen/cpp.py

```diff
@@ -2888,7 +2902,12 @@ def store(self, name, index, value, mode=None):
 
     def reduction(self, dtype, src_dtype, reduction_type, value):
         if not (
-            (dtype == torch.float and src_dtype == torch.float)
+            (
+                dtype == torch.bool
+                and src_dtype == torch.bool
+                and reduction_type == "any"
```

Comment:
Can we assume that when `reduction_type == "any"`, both dtype and src_dtype are `torch.bool`?

Reply:
The `dtype` is always `torch.bool` from doc https://pytorch.org/docs/stable/generated/torch.any.html#torch-any.
For the `src_dtype`, we only met cases with `torch.bool` with dynamo benchmark models. So I think we may can just assume for `bool` dtype and we can further support `src_dtype != torch.bool` if we meet this usage.

Reply:
I removed the check for `dtype` but keep the check for `src_dtype`.

> Just add "any" to the `VECTORIZABLE_RTYPES` would be enough?
I think it depends on whether we want an runtime error or just a log for `Disabled Vec` when we meet new case (`any` with `src_dtype != bool`). I feel it is a little bit aggressive to directly raise errors. Although we do net meet such a case in dynamo benchmarks.

Reply:
I think if it is supposed to be bool, we'd better check with assertion, not "if"... What do you think?

Reply:
Sure, I updated to only check `any` in `VECTORIZABLE_RTYPES`.
And there is already an assert here to raise error,  https://github.com/pytorch/pytorch/pull/131017/files#diff-5ab7b0235e2076a5fc6629ba0b109208940f5b94f5c13babc3e0f87cf4fcec82L2400